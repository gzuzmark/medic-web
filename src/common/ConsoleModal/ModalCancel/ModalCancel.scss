.ModalCancel {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  &_texts {
    margin: auto 0;
    text-align: center;
  }
  &_buttons {
    display: flex;
    justify-content: space-around;
    margin: 0 auto;
    width: 60%;
  }
}
