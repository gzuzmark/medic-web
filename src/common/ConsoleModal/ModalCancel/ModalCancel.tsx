import * as React from 'react';
import {ButtonNormal, THEME_SECONDARY} from "../../Buttons/Buttons";
import {Heading3, LIGHT_TEXT, Subhead1} from '../../MentorText';
import MentorModalBase from "../MentorModalBase";
import './ModalCancel.scss';

interface IPropsModalCancel {
    show: boolean;
    title: string;
    styles?: React.CSSProperties;
    disabled: boolean;
    loading: boolean;
    onCancel(): void;
    onConfirm(): void;
}

const ModalCancel: React.FC<IPropsModalCancel> = (props) => {
    const buttonProps = {};
    if (!!props.disabled) {
        buttonProps['disabled'] = true;
    } else if (!!props.loading) {
        buttonProps['loading'] = true;
    }
    return (
        <MentorModalBase show={props.show}>
            <div className={'ModalCancel'}>
                <div className='ModalCancel_texts'>
                    <Heading3 style={{margin: '7px 0'}}>Cancelar eliminar sesiones</Heading3>
                    <Subhead1 weight={LIGHT_TEXT} style={{margin: '7px 0'}}>Si cancelas, se perderá la información actual.</Subhead1>
                    <Subhead1 weight={LIGHT_TEXT} style={{margin: '7px 0'}}>¿Estás seguro?</Subhead1>
                </div>
                <div className='ModalCancel_buttons'>
                    <ButtonNormal attrs={{...buttonProps, onClick: props.onCancel}}
                                  type={THEME_SECONDARY} text="No, volver"/>
                    <ButtonNormal attrs={{...buttonProps, onClick: props.onConfirm}} text="Sí, cancelar"/>
                </div>
            </div>
        </MentorModalBase>
    );
};

export default ModalCancel;
