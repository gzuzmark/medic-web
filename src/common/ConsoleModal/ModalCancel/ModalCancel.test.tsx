import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ModalCancel from './ModalCancel';


describe('ModalCancel Test',() => {
    let props: any;
    let mountedModalCancel: any;
    const getComponent = () => {
        if (!mountedModalCancel) {
            mountedModalCancel = shallow(
                <ModalCancel {...props}>Contenido de la Modal</ModalCancel>
            );
        }
        return mountedModalCancel;
    };

    beforeEach(() => {
        props = {
            disabled: false,
            loading: false,
            onCancel: () => {
                //
            },
            onConfirm: () => {
                //
            },
            show: true,
            title: 'Titulo de la modal'
        };
        mountedModalCancel = undefined;
    });

    it("render: render ModalCancel", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });
});