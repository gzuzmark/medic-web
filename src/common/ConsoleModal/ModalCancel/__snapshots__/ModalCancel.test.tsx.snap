// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCancel Test render: render ModalCancel 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalCancel
    disabled={false}
    loading={false}
    onCancel={[Function]}
    onConfirm={[Function]}
    show={true}
    title="Titulo de la modal"
  >
    Contenido de la Modal
  </ModalCancel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="ModalCancel"
        >
          <styled.h2
            color="textBold"
          >
            Si cancelas, se perderá la información actual.
          </styled.h2>
          <styled.h2
            color="textBold"
            style={
              Object {
                "fontWeight": "normal",
              }
            }
          >
            ¿Estás seguro?
          </styled.h2>
        </div>,
        <ConfirmButtons
          cancelText="No, volver"
          confirmText="Sí, cancelar"
          disabled={false}
          loading={false}
          onCancel={[Function]}
          onConfirm={[Function]}
          styles={
            Object {
              "margin": "60px auto 20px auto",
              "width": "74%",
            }
          }
        />,
      ],
      "icon": "book",
      "show": true,
      "title": "Titulo de la modal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.h2
              color="textBold"
            >
              Si cancelas, se perderá la información actual.
            </styled.h2>,
            <styled.h2
              color="textBold"
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            >
              ¿Estás seguro?
            </styled.h2>,
          ],
          "className": "ModalCancel",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Si cancelas, se perderá la información actual.",
              "color": "textBold",
            },
            "ref": null,
            "rendered": "Si cancelas, se perderá la información actual.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "¿Estás seguro?",
              "color": "textBold",
              "style": Object {
                "fontWeight": "normal",
              },
            },
            "ref": null,
            "rendered": "¿Estás seguro?",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cancelText": "No, volver",
          "confirmText": "Sí, cancelar",
          "disabled": false,
          "loading": false,
          "onCancel": [Function],
          "onConfirm": [Function],
          "styles": Object {
            "margin": "60px auto 20px auto",
            "width": "74%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="ModalCancel"
          >
            <styled.h2
              color="textBold"
            >
              Si cancelas, se perderá la información actual.
            </styled.h2>
            <styled.h2
              color="textBold"
              style={
                Object {
                  "fontWeight": "normal",
                }
              }
            >
              ¿Estás seguro?
            </styled.h2>
          </div>,
          <ConfirmButtons
            cancelText="No, volver"
            confirmText="Sí, cancelar"
            disabled={false}
            loading={false}
            onCancel={[Function]}
            onConfirm={[Function]}
            styles={
              Object {
                "margin": "60px auto 20px auto",
                "width": "74%",
              }
            }
          />,
        ],
        "icon": "book",
        "show": true,
        "title": "Titulo de la modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <styled.h2
                color="textBold"
              >
                Si cancelas, se perderá la información actual.
              </styled.h2>,
              <styled.h2
                color="textBold"
                style={
                  Object {
                    "fontWeight": "normal",
                  }
                }
              >
                ¿Estás seguro?
              </styled.h2>,
            ],
            "className": "ModalCancel",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Si cancelas, se perderá la información actual.",
                "color": "textBold",
              },
              "ref": null,
              "rendered": "Si cancelas, se perderá la información actual.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "¿Estás seguro?",
                "color": "textBold",
                "style": Object {
                  "fontWeight": "normal",
                },
              },
              "ref": null,
              "rendered": "¿Estás seguro?",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cancelText": "No, volver",
            "confirmText": "Sí, cancelar",
            "disabled": false,
            "loading": false,
            "onCancel": [Function],
            "onConfirm": [Function],
            "styles": Object {
              "margin": "60px auto 20px auto",
              "width": "74%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
